File: ios/Classes/HardwareFeedbackManager.swift
import UIKit
import AudioToolbox

/// iOS Feedback Manager for Haptics and Vibrations.
public class HardwareFeedbackManager {
    public func vibrate(duration: Int) {
        AudioServicesPlaySystemSound(kSystemSoundID_Vibrate)
    }

    public func haptic(type: String) {
        switch type {
        case "impact":
            let generator = UIImpactFeedbackGenerator(style: .medium)
            generator.prepare()
            generator.impactOccurred()
        case "success":
            let generator = UINotificationFeedbackGenerator()
            generator.prepare()
            generator.notificationOccurred(.success)
        case "warning":
            let generator = UINotificationFeedbackGenerator()
            generator.prepare()
            generator.notificationOccurred(.warning)
        case "error":
            let generator = UINotificationFeedbackGenerator()
            generator.prepare()
            generator.notificationOccurred(.error)
        default:
            break
        }
    }
}

File: android/src/main/kotlin/com/example/nexora_sdk/HardwareFeedbackManager.kt
package com.example.nexora_sdk

import android.content.Context
import android.os.Build
import android.os.VibrationEffect
import android.os.Vibrator
import android.os.VibratorManager
import android.view.HapticFeedbackConstants

/**
 * Manages tactile hardware feedback including timed vibrations and pre-defined haptic patterns.
 */
class HardwareFeedbackManager(private val context: Context) {
    private val vibrator: Vibrator = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
        val manager = context.getSystemService(Context.VIBRATOR_MANAGER_SERVICE) as VibratorManager
        manager.defaultVibrator
    } else {
        context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
    }

    fun vibrate(duration: Long) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            vibrator.vibrate(VibrationEffect.createOneShot(duration, VibrationEffect.DEFAULT_AMPLITUDE))
        } else {
            vibrator.vibrate(duration)
        }
    }

    fun haptic(type: String) {
        // Pre-defined patterns using HapticFeedbackConstants for best system feel
        when (type) {
            "impact" -> vibrate(20)
            "success" -> vibrate(50)
            "warning" -> {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    vibrator.vibrate(VibrationEffect.createWaveform(longArrayOf(0, 50, 100, 50), -1))
                }
            }
            "error" -> {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    vibrator.vibrate(VibrationEffect.createWaveform(longArrayOf(0, 100, 50, 100), -1))
                }
            }
        }
    }
}
